src/main/webapp/js/common.js

```diff
@@ -839,3 +838,42 @@ function setChevronToDown(chevronContainer) {
     chevronContainer.addClass("glyphicon-chevron-down");
 }
 
+/**
+ * Changes the state of the panel (collapsed/expanded).
+ */
+function toggleSingleCollapse(e) {
+    if (!$(e.target).is('a') && !$(e.target).is('input')) {
+        var glyphIcon = $(this).find('.glyphicon');
+        var className = $(glyphIcon[0]).attr('class');
+        if (className.indexOf('glyphicon-chevron-up') != -1) {
+            hideSingleCollapse($(e.currentTarget).attr('data-target'));
+        } else {
+            showSingleCollapse($(e.currentTarget).attr('data-target'));
+        }
+    }
```

Comment:
If the major part of the method body is inside an `if`, that's a good place for using early returns (i.e. guard clauses) to reduce nesting. In general, the main flow of a method should be un-nested.